src/Qir/Execution/Tools/QirTools.cs

```diff
@@ -12,16 +16,41 @@ namespace Microsoft.Quantum.Qir.Tools
     /// </summary>
     public static class QirTools
     {
-
         /// <summary>
         /// Creates a QIR-based executable from a .NET DLL generated by the Q# compiler.
         /// </summary>
         /// <param name="qsharpDll">.NET DLL generated by the Q# compiler.</param>
         /// <param name="libraryDirectory">Directory where the libraries to link to are located.</param>
         /// <param name="includeDirectory">Directory where the headers needed for compilation are located.</param>
-        public static Task BuildFromQSharpDll(FileInfo qsharpDll, DirectoryInfo libraryDirectory, DirectoryInfo includeDirectory)
+        /// <param name="executablesDirectory">Directory where the created executables are placed.</param>
+        public static async Task BuildFromQSharpDll(
+            FileInfo qsharpDll,
+            DirectoryInfo libraryDirectory,
+            DirectoryInfo includeDirectory,
+            DirectoryInfo executablesDirectory)
         {
-            throw new NotImplementedException();
+            using var qirContentStream = new MemoryStream();
+            if (!AssemblyLoader.LoadQirByteCode(qsharpDll, qirContentStream))
+            {
+                throw new ArgumentException("The given DLL does not contain QIR byte code.");
+            }
+
+            if (!executablesDirectory.Exists)
+            {
+                executablesDirectory.Create();
+            }
+
+            var dirStr = executablesDirectory.FullName + Path.DirectorySeparatorChar;
```

Comment:
With the suggestion using `Path.Combine`, this variable is no longer needed.

Reply:
removed